### DISCUSSION TITLE
Conflict between Code annotations and markdown-exec lib

### DISCUSSION THREAD

**Discussion by gabrielbdornas** [squidfunk/mkdocs-material#7302] [Q&A]
### Context

I'm trying to build a series of python tutorials and think it will be nice to have the result of the code block execution in a result table. For this, I found the [markdown-exec](https://pypi.org/project/markdown-exec/) to help me.

I'm my current project, as I'm working in tutorials, I've been using [code annotations](https://squidfunk.github.io/mkdocs-material/reference/code-blocks/?h=code+annotations#code-annotations) a lot.

After install and configuring [markdown-exec](https://pypi.org/project/markdown-exec/) within `markdown_extensions` property, the code annotations stopped working.

So, the code results are working, but the cool code annotations option crash:

![image](https://github.com/squidfunk/mkdocs-material/assets/49699290/22e20170-12ba-4d0a-a672-0aaf4350b9a2)

![image](https://github.com/squidfunk/mkdocs-material/assets/49699290/23a506b1-5941-40a1-b233-c336343a41b7)

### Bug description

To me seems a kind of conflict betwwen the [code annotations](https://squidfunk.github.io/mkdocs-material/reference/code-blocks/?h=code+annotations#code-annotations) config and the [markdown-exec](https://pypi.org/project/markdown-exec/).

- When I do not add `markdown-exec` options to my code block, the annotations works:

![image](https://github.com/squidfunk/mkdocs-material/assets/49699290/c8f07034-9b88-4789-9a56-8f6a5d2f41a5)

![image](https://github.com/squidfunk/mkdocs-material/assets/49699290/82746a57-bb47-4fc3-8d8d-767751e71b21)

- When add `markdown-exec` options to my code block, it starts to work, but the annotations crash (`exec="1" source="tabbed-left"`):

![image](https://github.com/squidfunk/mkdocs-material/assets/49699290/0a85080a-fa0e-4dab-9da4-e277735fba32)

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
- [FR: Editable code blocks issue](https://github.com/squidfunk/mkdocs-material/issues/5527).
- [Run PythonCode on a docu side discussion](https://github.com/squidfunk/mkdocs-material/discussions/4649).

### Reproduction

The problem could be seen at this [branch](https://github.com/gabrielbdornas/code/tree/17/code-block-results) of my code.

### Steps to reproduce

1. [Clone](https://github.com/gabrielbdornas/code/tree/17/code-block-results) the code.
2. Install requirements with `pip install -r requirements.txt`.
3. Rum `mkdocs serve`.
4. Viste the page located at `http://localhost:8000/blog/python-course---class-1/#your-first-python-program` 

Obs.: The conflicted code is located at [this file](https://github.com/gabrielbdornas/code/blob/072e32190a1df34e360799cd2ce5e7d6a4aff0a0/docs/blog/posts/20240626_python_course.md?plain=1#L33-L37).

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [ ] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) using the [built-in info plugin](https://squidfunk.github.io/mkdocs-material/plugins/info/).

**Comment by squidfunk** [comment]
Thanks for reporting. Since we do not officially support `markdown-exec`, I'm converting this to a discussion and tagging @pawamoy for visibility. If we can make it work, happy to review a PR that adds the relevant changes, but we're currently frying much, much bigger fish that we cannot budget time to add support for additional third-party libraries 🤟

**Reply by squidfunk** [reply]
Also note that providing a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) is __absolutely essential__ for us and other maintainers to look into it. You haven't provided one but ticked the box. Please provide one, so that somebody can find some motivation and time to help you.

**Reply by gabrielbdornas** [reply]
> Thanks for reporting. Since we do not officially support `markdown-exec`, I'm converting this to a discussion and tagging @pawamoy for visibility. If we can make it work, happy to review a PR that adds the relevant changes, but we're currently frying much, much bigger fish that we cannot budget time to add support for additional third-party libraries 🤟

@squidfunk, appreciate your time and effort to answer this so quickly. One more time, I want to take advantage of the opportunity to say that the lib is outstanding! Thank you all so much!

**Comment by pawamoy** [comment]
The issue is probably the following:

- markdown-exec reads your code block and executes it
- then it prepares Markdown markup that wraps both the source and the result in tabs
- then it renders everything and returns that (HTML)
- in the end, your bullet list is not right after your code block anymore: the code block is within the tab structure, but the bullet list is outside
- MfM only supports code annotations when the bullet list is right after the code block
- therefore the code annotations logic breaks

I don't see a solution yet, will think about it.

**Reply by squidfunk** [reply]
Great analysis and explanation, thanks! I guess both features kind of overlap then, and this particularly becomes a problem because they are mounted/executed at different stages.
